iModelCore/ECPresentation/Tests/NonPublished/Unit/Queries/CustomFunctionTests.cpp

```diff
@@ -234,12 +233,272 @@ TEST_F(CustomFunctionTests, EvaluateECExpression)
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.BindInt(2, (int)PRIMITIVETYPE_String));
+    ASSERT_TRUE(DbResult::BE_SQLITE_ROW == stmt.Step());
+    ASSERT_STREQ("12345678987", stmt.GetValueText(0));
+
+    stmt.Finalize();
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.Prepare(GetDb(), "SELECT EvaluateECExpression(ECClassId, ECInstanceId, ?, ?) FROM RET.Widget"));
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.BindText(1, booleanExpression, IECSqlBinder::MakeCopy::No));
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.BindInt(2, (int)PRIMITIVETYPE_String));
+    ASSERT_TRUE(DbResult::BE_SQLITE_ROW == stmt.Step());
+    ASSERT_STREQ("True", stmt.GetValueText(0));
+
+    stmt.Finalize();
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.Prepare(GetDb(), "SELECT EvaluateECExpression(ECClassId, ECInstanceId, ?, ?) FROM RET.Widget"));
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.BindText(1, doubleExpression, IECSqlBinder::MakeCopy::No));
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.BindInt(2, (int)PRIMITIVETYPE_String));
+    ASSERT_TRUE(DbResult::BE_SQLITE_ROW == stmt.Step());
+    ASSERT_STREQ("2.5", stmt.GetValueText(0));
+
+    stmt.Finalize();
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.Prepare(GetDb(), "SELECT EvaluateECExpression(ECClassId, ECInstanceId, ?, ?) FROM RET.Widget"));
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.BindText(1, dateTimeExpression, IECSqlBinder::MakeCopy::No));
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.BindInt(2, (int)PRIMITIVETYPE_String));
+    ASSERT_TRUE(DbResult::BE_SQLITE_ROW == stmt.Step());
+    ASSERT_STREQ("2024-09-12T00:00:00.000", stmt.GetValueText(0));
+
+    // verify it works when converting to int
+    stmt.Finalize();
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.Prepare(GetDb(), "SELECT EvaluateECExpression(ECClassId, ECInstanceId, ?, ?) FROM RET.Widget"));
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.BindText(1, stringExpression, IECSqlBinder::MakeCopy::No));
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.BindInt(2, (int)PRIMITIVETYPE_Integer));
+    ASSERT_TRUE(DbResult::BE_SQLITE_ERROR == stmt.Step());
+    ASSERT_STREQ(errorMessage, stmt.GetDataSourceDb()->GetLastError().c_str());
+
+    stmt.Finalize();
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.Prepare(GetDb(), "SELECT EvaluateECExpression(ECClassId, ECInstanceId, ?, ?) FROM RET.Widget"));
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.BindText(1, intExpression, IECSqlBinder::MakeCopy::No));
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.BindInt(2, (int)PRIMITIVETYPE_Integer));
+    ASSERT_TRUE(DbResult::BE_SQLITE_ROW == stmt.Step());
+    ASSERT_EQ(4, stmt.GetValueInt(0));
+
+    stmt.Finalize();
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.Prepare(GetDb(), "SELECT EvaluateECExpression(ECClassId, ECInstanceId, ?, ?) FROM RET.Widget"));
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.BindText(1, longExpression, IECSqlBinder::MakeCopy::No));
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.BindInt(2, (int)PRIMITIVETYPE_String));

```

Comment:
should this use `PRIMITIVETYPE_Integer`?

Reply:
Yes, changed